// semantic-ui-calendar-react/dist/views/CalendarHeader/HeaderWeeks.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _semanticUiReact = require('semantic-ui-react');

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** Return array of week day names.
 * 
 * getWeekDays() --> ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Su']
 */
var getWeekDays = function getWeekDays() {
  var long = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;

  var weekDays = [];
  var day = (0, _moment2.default)().startOf('week');
  for (var i = 0; i < 7; i++) {
    weekDays[i] = day.format(long ? 'dddd' : 'dd');
    day.add(1, 'd');
  }
  return weekDays;
};

var cellStyle = {
  border: 'none',
  borderBottom: '1px solid rgba(34,36,38,.1)'
};

var weekDayCells = getWeekDays().map(function (weekDay) {
  return _react2.default.createElement(
    _semanticUiReact.Table.HeaderCell,
    {
      key: weekDay,
      style: cellStyle,
      colSpan: '1' },
    weekDay
  );
});

function HeaderWeeks() {
  return _react2.default.createElement(
    _semanticUiReact.Table.Row,
    null,
    weekDayCells
  );
}

HeaderWeeks.handledProps = [];
exports.default = HeaderWeeks;
